docs/internal_abi.md

```diff
@@ -0,0 +1,323 @@
+That said, it is useful for people working on the compiler, producing additional tooling, and writing tests for the ABI used internally to be documented.
+This draft is a proposal for the v1 ABI.
+
+Primitive types
+---------------
+
+The primitive types in Verona are:
+
+ - Boolean
+ - Integers, signed or unsigned, with 1-128 bits of precision
+ - Floating point values, with various power-of-two sizes
+ - Pointers to objects
+
+The representation of these may change when used in a complex type expression but in the absence of other constraints they will be stored as simple values, as described in the following sections.
+The compiler is free to relax any of these constraints for on-stack variables that are not passed by reference (for example, leaving some bits in a value undefined if the compiler can prove that they are never read).
+
+Verona places the following restrictions on target architectures:
+
+ - The size of a byte is 8 bits.
+ - The architecture can load and store a single byte without read-modify-write operations
+ - The order of bits within different-sized values is consistent
+ - Signed arithmetic on integers uses twos complement
+
+This is a slightly tighter set of requirements than the C specification but is looser than the de-facto C standard as used in large deployed codebases.
+
+### Booleans
+
+Boolean values are stored as bytes, with the bit pattern 0 or 1 in the platform's native endian.
+The compiler is responsible for ensuring that no bits other than the least-significant bit in a boolean are ever non-zero.
+
+The alignment of booleans is 1 byte.
+
+### Integers
+
+Power-of-two-sized integers in Verona can be used for arithmetic and so are expected to use the representation that the target architecture uses for integer operations.
+The types `U8`, `U16`, `U32`, `U64`, and `U128` are all unsigned fixed-width integers with the platform's native endian.
+The types `S8`, `S16`, `S32`, `S64`, and `S128` are all signed fixed-width integers represented as twos complement numbers with the platform's native endian.
+These types are always stored naturally aligned.
+For example, a `U64` is always stored on an 8-byte boundary.
+
+Other integer types, for example `U56` are *storage-only* types.
+These must be extended to a power-of-two sized type for arithmetic.
+The value of these types is primarily for use in structures or union types, where special layout rules apply. Their storage semantics are explained below.
+When stored on the stack, however, these should also be sign- or zero-extended (for signed and unsigned types, respectively) to the next native integer type.
+
+### Floating-point values
+
+The types `Float16`, `Float32`, `Float64` and `Float128` correspond to IEEE 754 binary16, binary32, binary64, and binary128 types, respectively.
+*Note:* A future version of the language may provide IEEE decimal types, binary256, bfloat32, or smaller floating point types but these are not necessary for MVP.
```

Comment:
What's `MVP`?

Reply:
Minimum Viable Product.  The pre-1.0 release that's complete enough for folks to actually try writing real code in Verona.

Reply:
Right, then perhaps write that explicitly, as I keep looking at it and thinking of minimum / maximum and the acronym completions make no sense. :)